src/SFA.DAS.FAA.Data/SavedSearch/SavedSearchRepository.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.FAA.Domain.Entities;
+using SFA.DAS.FAA.Domain.Models;
+
+namespace SFA.DAS.FAA.Data.SavedSearch
+{
+    public interface ISavedSearchRepository
+    {
+        Task<PaginatedList<SavedSearchEntity>> GetAll(DateTime dateFilter, int pageNumber, int pageSize, CancellationToken token);
+    }
+
+    public class SavedSearchRepository(IFindApprenticeshipsDataContext dataContext) : ISavedSearchRepository
+    {
+        public async Task<PaginatedList<SavedSearchEntity>> GetAll(DateTime dateFilter, int pageNumber, int pageSize, CancellationToken token = default)
+        {
+            // Query
+            var query = dataContext
+                .SavedSearchEntities
+                .Where(fil => fil.LastRunDate == null || fil.LastRunDate > dateFilter);
```

Comment:
Would it be clearer to pass 'days' rather than dateFilter? So I'm querying for records where lastrun is more than x days ago. Instead of having to calculate the date that would be outside the repository.

Reply:
I believe it's the responsibility of the caller to define the filter, there could be a use case for a new job to fetch recent saved searches and send special alerts coming up in the future, in that case if the filter logic sits inside repo then we have to touch base/alter the logic in the repo, which is not idle.